### ISSUE TITLE
Rule Change: allow just 'consistent' in curly

### ISSUE THREAD

**Issue by asselin** [eslint/eslint#20215]
### What rule do you want to change?

curly

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

Other

### Example code

```js
if (count == 0) {
  console.log(1);
} else console.log(2);
```

### What does the rule currently do for this code?

Today, curly requires specifying `all`, `multi`, `multi-line`, or `multi-or-nest` in order to specify `consistent` (`consistent` can't be specified on its own).  What I'd like is a way to have eslint not enforce anything about when to use braces, JUST to enforce that if they're used on the `if()` they also need to be on the `else`, and vice versa.

### What will the rule do after it's changed?

- Change the options validator to allow specifying just `consistent` without also needing to also specify `all`, `multi`, `multi-line`, or `multi-or-nest`
- If just `consistent` is specified, add a case to `prepareCheck` https://github.com/eslint/eslint/blob/004577eda2f2f4b2829e0364f8b41893cebfc859/lib/rules/curly.js#L193 that returns `expected = null`.

After this change, only the `consistent` part of the rule will flag issues, and the code above will be flagged:

**Bad**
```
if (count == 0) {
  console.log(1);
} else console.log(2);
```

**Good**
```
if (count == 0) {
  console.log(1);
} else { 
   console.log(2); 
}
```

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @asselin! Thank for the proposal, but as stated in the [documentation](https://eslint.org/docs/latest/rules/curly), this rule is frozen and is not accepting feature requests. You can always copy a core rule into a custom rule and modify it the way you want.

**Comment by asselin** [comment]
@mdjermanovic I wish you'd reconsider your response on this.  Prettier enforces a brace style that does not match any of the options allowed by this rule, so if you want to just use prettier's rules, curly has to be disabled.  However, what prettier does not enforce is making all branches of if() either have or not have braces.  curly has the ability to do that, but it presently requires you to also select one of the 4 brace styles too.  Changing about 20 lines, to allow just enabling the `consistent` portion of this rule would provide that functionality.  It would not introduce any new switches or flags on the rule, and not affect any existing usage; it would simply allow ALSO using this rule in additional contexts.  If seeing the PR would help, LMK and I'll open one.
